src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -20,7 +22,8 @@ public class CodeGenIL : IDisposable
         internal string className;
         internal string methodName;
         private IDictionary<string, IList> input;
-
+        public ProtoCore.MSILRuntimeCore runtimeCore;
```

Comment:
does this need to be public?

Reply:
It does not. A lot of the new APIs do not need to be.
Making it here public made it easier to test out.

I can either make everything internal (or most of it)  ...or leave it as is and create a separate jira task for it